Setup Code and CodeGroup with initial SQL data

shawnchoi8 opened the pull request:
## Summary (요약)
- Completed logic related to Code and CodeGroup
- Created code-data.sql to initialize Code and CodeGroup tables with default data.

- User entity, repository, and service with code-related adjustments
  - Renamed methods in User domain to align with Code/CodeGroup changes
  - Completed initial code setup logic for User

## Type of change
- [ ] Bug fix (버그 수정)
- [x] New feature (새로운 기능 추가)
- [x] Refactoring (리팩토링)
- [ ] Documentation (문서화)
- [ ] Tests (테스트 추가/수정)

---

## 📌 PR Checklist
- [x] 브랜치 이름 규칙 확인 (feature/*, fix/* 등)
- [x] PR 대상 브랜치 확인 (develop)
- [x] 최신 develop 반영 완료 (merge or rebase)
- [x] 코드 자동 정렬 실행 (⌥⌘L / Ctrl+Alt+L) & 불필요한 import 제거
- [x] 로컬 빌드/테스트 확인 완료

jaeaeee committed:
feat: add refund

shawnchoi8 committed:
feat: Integrate Spring Security with initial SecurityConfig setup

shawnchoi8 committed:
refactor: Code and CodeGroup entities - rename variables and add business methods

shawnchoi8 committed:
feat: CodeRepository, CodeGroupRepository

jaeaeee committed:
add

shawnchoi8 committed:
feat: CodeRepository, CodeGroupRepository, CodeService, CodeController

jaeaeee committed:
add

jaeaeee committed:
add

jaeaeee committed:
remove CodeFinder and CodeRepository

jaeaeee committed:
feat: replace CodeFinder with CodeService numeric lookups

shawnchoi8 committed:
feat: add code-data.sql file for data injection,

shawnchoi8 committed:
Merge branch 'feature/ordering' into feature/security

shawnchoi8 committed:
Merge dependencies: keep springdoc from develop and add security

shawnchoi8 committed:
refactor: Remove unnecessary data.sql

shawnchoi8 committed:
feat: configure JPA to initialize tables and load code-data.sql (set up application.yaml to run code-data.sql automatically after JPA table creation)

soojjung reviewed (changes requested):
`User.java` 파일에서 에러 발생합니다. 에러 해결해주세요.
그리고 코드 자동 정렬 실행 `(⌥⌘L / Ctrl+Alt+L)` & 불필요한 import 제거도 확인 부탁드립니다.

shawnchoi8 committed:
refactor: Refactor User entity, repository, and service with code-related adjustments

- Renamed methods in User domain to align with Code/CodeGroup changes
- Completed initial code setup logic for User

shawnchoi8 commented:
> `User.java` 파일에서 에러 발생합니다. 에러 해결해주세요. 그리고 코드 자동 정렬 실행 `(⌥⌘L / Ctrl+Alt+L)` & 불필요한 import 제거도 확인 부탁드립니다.

에러 수정 완료했고 자동정렬도 완료했습니다.

shawnchoi8 committed:
chore: remove leftover TODO comment, clean up redundant fetch = LAZY option in non-owning @OneToOne

shawnchoi8 committed:
refactor: basic Spring Security configuration

- Allow public access to API endpoints
- Add root path (/) access permission
- Complete basic security setup before JWT implementation

shawnchoi8 committed:
refactor: SecurityConfig - add path("/swagger-ui/**", "/v3/api-docs/**") for swagger test

soojjung reviewed (approved):
수고 하셨습니다.
